### ISSUE TITLE
MascotAdapterOnline seems not to work with defaults

### ISSUE THREAD

**Issue by bernt-matthias** [OpenMS/OpenMS#4412]
When running `MascotAdapterOnline` with defaults its not working: `ending run with Mascot Server replied: 'Host  not found'`. Would it be an idea to set argument default values such that it works with the/a public one? E.g. the parameters used in the test (https://github.com/OpenMS/OpenMS/blob/ce53d13da1fc6f499f4cb0eb7de28a0bdb6e8740/src/tests/topp/THIRDPARTY/third_party_tests.cmake#L200) `-Mascot_parameters:email 'openmsjenkins@gmail.com' -Mascot_server:hostname www.matrixscience.com -Mascot_server:host_port '80' -Mascot_server:server_path ''`

As you might know I'm working on the Galaxy wrappers for OpenMS. I'm wondering if I should set any parameters specifically? Default server, mail, username, ...?

**Comment by jpfeuffer** [comment]
This is a bit tricky since this is an online test server. I think you should sign up with your own credentials and check out things like proxies, firewalls etc.

**Comment by bernt-matthias** [comment]
Are those parameters optional (couldn't find where they are defined)? Then it might be good to make them mandatory.

**Comment by bernt-matthias** [comment]
Stubled again over this adapter again in 3.2: 

`TOPP_MascotAdapterOnline_1` fails with:

``` failed
    /opt/conda/conda-bld/openms-meta_1728651094919/work/src/openms/source/FORMAT/HANDLERS/XMLHandler.cpp(105): While loading '/tmp/20241124_130757_MOLSYB009_510036_1_Mascot_response': expected end of tag 'LINK'( in line 7 column 3)
    Error: Unable to read file (While loading '/tmp/20241124_130757_MOLSYB009_510036_1_Mascot_response': expected end of tag 'LINK'( in line 7 column 3) in: /tmp/20241124_130757_MOLSYB009_510036_1_Mascot_response)
stdout:
Progress of 'loading mzML':
  Progress of 'loading spectra list':
  -- done [took 0.01 s (CPU), 0.00 s (Wall)] -- 
-- done [took 0.01 s (CPU), 0.00 s (Wall) @ 10.67 MiB/s] -- 
Number of spectra loaded: 3
MascotGenericFile: no native ID accession.
Submitting Mascot query (now: 2024-11-24 13:07:51)...
Mascot query finished
```

Response file is:

```
<!DOCTYPE html>
<HTML>
<HEAD>
<META HTTP-EQUIV="X-UA-Compatible" CONTENT="IE=edge">
<LINK REL="shortcut icon" href="../favicon.ico">
<TITLE>Fatal Error</TITLE>
</HEAD>
<BODY>
<H1>Fatal Error</H1>
<P><B>no file name</B>
</BODY></HTML>
```

executing `TOPP_MascotAdapterOnline_2`

Just does nothing.

**Comment by jpfeuffer** [comment]
Hi! What was the full call? Don't you have to give it a file?
By the way, the response indeed seems to be malformed. Link has to close somewhere. This part probably has to be fixed on the mascot side.

**Comment by jpfeuffer** [comment]
Seems like HTML allows self-closing tags for some tag types.
But we kind of require XHTML since we are using a default XML parser.

**Comment by jpfeuffer** [comment]
Anyway, this error page should not even be the response if everything is correct.

**Comment by bernt-matthias** [comment]
I execute (should be the same as in the test CMake files): 

```
MascotAdapterOnline -test -ini MascotAdapterOnline_1.ini -Mascot_parameters:database SwissProt -in spectra_comet.mzML -out MascotAdapterOnline_1_out1.tmp.idXML > TOPP_MascotAdapterOnline_1.stdout 2> TOPP_MascotAdapterOnline_1.stderr
MascotAdapterOnline -test -ini MascotAdapterOnline_1.ini -debug 666 -Mascot_parameters:decoy -Mascot_parameters:database SwissProt -in spectra_comet.mzML -out MascotAdapterOnline_2_out1.tmp.idXML > TOPP_MascotAdapterOnline_2.stdout 2> TOPP_MascotAdapterOnline_2.stderr
```

**Comment by jpfeuffer** [comment]
Hmm sorry I am not too familiar with Mascot Online. Not even sure if the public server still exists or if the API changed.

I think it is rarely used and I am not sure if we can invest a lot of time debugging/supporting this without help from interested people.

One would need to check the responses from the server. Their API. Run the tests with debug. Inspect the files being sent and the responses that are received.

As mentioned in the beginning of this issue, you should probably sign up yourself and add all your own configuration options.

**Comment by jpfeuffer** [comment]
Also try to send the files and receive responses via curl or wget outside of the adapter.

**Comment by bernt-matthias** [comment]
> I am not sure if we can invest a lot of time debugging/supporting this without help from interested people.

Same here. I will remove the Galaxy tool from updates until this is resolved.

**Comment by bernt-matthias** [comment]
The `-debug`  option does not add much information to do a quick check. 

> I am not sure if we can invest a lot of time debugging/supporting this without help from interested people

Same here. I'm will remove Galaxy support for the time being.

**Comment by jpfeuffer** [comment]
Yes, reasonable. If someone complains, we can still invest more time.
